tests/sklearn/test_sklearn_models.py

```diff
@@ -1055,7 +1038,49 @@ def check_exposition_structural_methods_decision_trees(model, x, y):
     )
 
 
-@pytest.mark.parametrize("model_class, parameters", sklearn_models_and_datasets)
+def check_load_fitted_sklearn_linear_models(model_class, n_bits, x, y):
+    """Check that linear models and QNNs support loading from pre-trained scikit-learn models."""
+
+    model = instantiate_model_generic(model_class, n_bits=n_bits)
+
+    # Fit the model and retrieve both the Concrete ML and the scikit-learn models
+    with warnings.catch_warnings():
+        # Sometimes, we miss convergence, which is not a problem for our test
+        warnings.simplefilter("ignore", category=ConvergenceWarning)
+        concrete_model, sklearn_model = model.fit_benchmark(x, y)
+
+    # This step is needed in order to handle partial classes
+    model_class = get_model_class(model_class)
+
+    # Load a Concrete ML model from the fitted scikit-learn one
+    loaded_concrete_model = model_class.from_sklearn_model(sklearn_model, X=x, n_bits=n_bits)
+
+    # Compile both the initial Concrete ML model and the loaded one
+    concrete_model.compile(x)
+    loaded_concrete_model.compile(x)
+
+    # Compute and compare the predictions from both models
+    y_pred_simulate = concrete_model.predict(x, fhe="simulate")
+    y_pred_simulate_loaded = loaded_concrete_model.predict(x, fhe="simulate")
+
+    assert numpy.isclose(y_pred_simulate, y_pred_simulate_loaded).all(), (
+        "Simulated predictions from the initial model do not match the ones made from the "
+        "loaded one."
+    )
+
+
+# Neural network models are skipped for this test
+# The `fit_benchmark` function of QNNs returns a QAT model and a FP32 model that is similar
+# in structure but trained from scratch. Furthermore, the `n_bits` setting
+# of the QNN instantiation in `instantiate_model_generic` takes `n_bits` as
+# a target accumulator and sets 3-b w&a for these tests. Thus it's
+# impossible to reach R-2 of 0.99 when comparing the two NN models returned by `fit_benchmark`
+@pytest.mark.parametrize(
+    "model_class, parameters",
+    get_sklearn_linear_models_and_datasets()
```

Comment:
could be replaced by `get_sklearn_all_models_and_datasets(exclude="NeuralNet")`, not sure what's best